src/components/TodoList/TodoList.tsx

```diff
@@ -1,208 +1,56 @@
-/* eslint-disable max-len */
-import React from 'react';
+import { useEffect, useState } from 'react';
+import { getTodos } from '../../api';
+import { Loader } from '../Loader/Loader';
+import { TodoHead } from './TodoHead';
+import { TodoItem } from './TodoItem';
+import { Todo } from '../../types/Todo';
+import { filterBy } from '../../utils/filterBy';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setTodo } from '../../features/todos';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
+  const todos = useAppSelector(state => state.todos);
+  const [list, setList] = useState<Todo[]>([]);
+  const [isProcesing, setIsProcesing] = useState(false);
+
+  useEffect(() => {
+    setIsProcesing(true);
+    getTodos()
+      .then(response => {
+        setList(response);
+        dispatch(setTodo(response));
+      })
+      .finally(() => setIsProcesing(false));
+  }, []);
+
+  useEffect(() => {
+    dispatch(setTodo(filterBy(list, query, status)));
```

Comment:
The `filterBy` function is called within the `useEffect` hook, but the function is not listed in the dependency array. This may cause unexpected behavior. Consider adding `filterBy` and `list` to the dependency array of the `useEffect` hook.